gossip/src/legacy_contact_info.rs

```diff
@@ -50,6 +55,30 @@ impl Sanitize for LegacyContactInfo {
     }
 }
 
+macro_rules! get_socket {
+    ($name:ident) => {
+        pub fn $name(&self) -> Result<SocketAddr, Error> {
+            let socket = &self.$name;
+            sanitize_socket(socket)?;
```

Comment:
We didn't used to sanitize on every get, right? I'm assuming the impact of this is negligible. The unwrap() on every get is kind of ugly though.. Do we ever expect this to fail? Could we perform some unwrap/expect within the getter and only return the socket?

Reply:
> We didn't used to sanitize on every get, right?

The sensitization was done later using `ContactInfo::is_valid_address`. 

> The unwrap() on every get is kind of ugly though..

There shouldn't be any unwraps except in tests.

> Could we perform some unwrap/expect within the getter and only return the socket?

We can't. We may receive contact-infos from other nodes with invalid sockets; and that is technically fine.

Reply:
> The sensitization was done later using ContactInfo::is_valid_address

Yup, and to confirm, it looks like `sanitize_socket()` doesn't do the `SocketAddrSpace` check that `ContactInfo::is_valid_address()` previously did via `LegacyContactInfo::is_valid_address()` , but you added the address space check in those call locations. Mentioning that here as I noticed it, and will make a pass at ensuring that all call sites are equivalent.

Reply:
> We didn't used to sanitize on every get, right? I'm assuming the impact of this is negligible. The unwrap() on every get is kind of ugly though.. Do we ever expect this to fail? Could we perform some unwrap/expect within the getter and only return the socket?

you may find this pr informative(?) and entertaining

https://github.com/solana-labs/solana/pull/30022